token-locker/src/lib.rs

```diff
@@ -326,6 +363,7 @@ impl Contract {
         required_deposit
     }
 
+    /// Update Ethereum Address of the Bridge Token Factory
     #[private]
     pub fn update_factory_address(&mut self, factory_address: String) {
         self.eth_factory_address = validate_eth_address(factory_address);
```

Comment:
I think `validate_eth_address` is not a clear name, it is more like a constructor for EthAddress from String. I think the name `eth_address_from_string` is more clear. I think it is Ok to do some checks in the constructor. I expect that the function `validate_eth_address` return is this address is valid or not.